frame/contracts/src/weights.rs

```diff
@@ -150,1183 +150,1191 @@ pub trait WeightInfo {
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	fn on_initialize() -> Weight {
-		(3_659_000 as Weight)
+		(3_972_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 	}
 	fn on_initialize_per_trie_key(k: u32, ) -> Weight {
-		(40_731_000 as Weight)
-			// Standard Error: 4_000
-			.saturating_add((2_317_000 as Weight).saturating_mul(k as Weight))
+		(44_890_000 as Weight)
+			// Standard Error: 5_000
+			.saturating_add((2_325_000 as Weight).saturating_mul(k as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(k as Weight)))
 	}
 	fn on_initialize_per_queue_item(q: u32, ) -> Weight {
-		(384_459_000 as Weight)
-			// Standard Error: 45_000
-			.saturating_add((146_401_000 as Weight).saturating_mul(q as Weight))
+		(268_333_000 as Weight)
+			// Standard Error: 32_000
+			.saturating_add((155_020_000 as Weight).saturating_mul(q as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn update_schedule() -> Weight {
-		(27_803_000 as Weight)
+		(27_452_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
-	fn put_code(n: u32, ) -> Weight {
+	fn instantiate_with_code(c: u32, n: u32, s: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 208_000
-			.saturating_add((110_774_000 as Weight).saturating_mul(n as Weight))
-			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			// Standard Error: 95_000
+			.saturating_add((154_745_000 as Weight).saturating_mul(c as Weight))
+			// Standard Error: 47_000
+			.saturating_add((370_000 as Weight).saturating_mul(n as Weight))
```

Comment:
I have no idea why `n` which is the size of the input data in kilobytes carries 100x the weight when supplied to `instantiate_with_code` (370_000) vs `instantiate` (4_000). In both cases the data is just passed to the constructor and nothing else. Maybe some else has an idea @thiolliere

Reply:
`n` has basically no influence on the runtime (it is too small to measure). So this difference is basically just random noise. It used to had an influence when the data was hashed for the contract address. However, this is no longer the case as `salt` is now the parameter which is hashed. I am removing those parameters.